### DISCUSSION TITLE
PEES for iStore (Australian Users)

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [JensenNick/huawei_solar_pees#44] [Forum]
@JensenNick Hi Nick,

After working on the Huawei WLCRS + SunSynk card, I was getting asked to make sensors for tracking all the 'PV to House' / 'PV to Grid' etc flows. Took a step back and thought there should be a 'default' set of custom sensors published for all WLCRS users.
Whilst I've done the ones that allow things like combining the readings / status / temperatures etc of devices into a single sensor for each (i.e. Inv 1 Active Power + Inv 2 Active Power = InverterS Active Power), I'm thinking it makes no sense to write 'another' set of sensors.

Given this wondering if it would be OK to fork PEES and then customise it for Australia, where we now have iStore (that has taken over from Huawei, under an OEM agreement). If this is OK, I'd then be looking to customise it to support the Australian market and Energy Retailers plans (probably using one local provider with an API of live import/export rates) and the more tradtional 'Fixed p/kWh + Daily fee' and 'Time of Use (ToU)' based pricing we have.

Wondering this would be OK, and if so, if it would be OK to query some Danish language and how the current logic works to modify to Australia?

TIA.

**Comment by JensenNick** [comment]
@Roving-Ronin By all means be my guest and I'd like to se what you come up with.

**Reply by Roving-Ronin** [reply]
@JensenNick Hi Nick,

Just wondering:

a) For the STATs package, is there any cards or graphs etc? Just wondering where it's published outputs are?

b) Is there any issues with people using this if they don't have a LUNA battery? (If not, this OK to change 'battery_rated_capacity  --> min:' from 5 to 0 ?

c) Have you considered adding template sensors for the DKK / $ sensors, to have them publish the DKK / $ at the front of the number? Or is this a $ users only problem, as we normally write it as '$ 123.00', but DKK is normally written as  '123,00 DKK' ?

i.e. in Developer --> Template, paste this example:

```
Economy - Expenses w PV  (Existing)
{{ states('sensor.economy_expenses_w_pv') }}

Economy - Expenses w PV  (Formatted)
value_template: "{{ '$ {:,.2f}'.format(states('sensor.economy_expenses_w_pv') | round(2)) }}"
```

Note formatting the sensor to prefix the unit value / symbol makes it a text type sensor, that can't then be used in things like graphs. So normally people end up doing a sensor like PEES has currently and adding a copy with the formating and different unique_id.

**Reply by JensenNick** [reply]
@Roving-Ronin 
Ad a) No, I haven't done much work in Lovelace. I probably should make a "brutto" list of the sensors included.
Ad b) The STAT package isn't limited to battery statistics, but off course if you have no battery, the battery efficiency will be useless. I'll have a look if it is causing any issue to set the battery capacity to 0.
Ad c) I don't think it is as simple to create a template sensor for currency (I might be wrong). I did try to create a currency selector in the input file, but it did not work with the sensors where the unit_of_measurement is a currency. I came to the conclusion, that the only right way to do this is to grab currency etc. from HA user settings, but that is beyond me. DKK is an accepted currency in HA and will work with graphs etc. but off course it is not ideal for non Danish users. I also had a look at the regex function, but not sure if I would be able to get that to work. Maybe the best (for now) would be if someone developed a "plugin" that could replace relevant data.
